ote/src/cljs/ote/views/service_search.cljs

```diff
@@ -147,64 +153,75 @@
                                   (stylefy/use-style style/simple-result-card-row))
                   (str (::t-service/name c) " (" (::t-service/business-id c) ")")])))))])))
 
-
-(defn- result-card [e! admin?
-                        {::t-service/keys [id name sub-type description
-                                           operator-name business-id transport-operator-id service-companies companies]
-                         :as service}]
+(defn- result-card
+  [e! admin?
+   {::t-service/keys [id name sub-type description transport-type homepage
+                      operator-name business-id transport-operator-id service-companies companies]
+    :as service}]
   (let [sub-type-tr (tr-key [:enums ::t-service/sub-type])
         e-links [external-interface-links service]
         service-desc (t-service/localized-text-for "FI" description)
+        formatted-desc (cond
+                         (> (count service-desc) 90)
+                         [:span (str (string/join (take 80 service-desc)) "... ")
+                          [common-ui/linkify (str "/#/service/" transport-operator-id "/" id)
+                           (tr [:service-search :show-all-information])]]
+                         (not-empty service-desc)
+                         [:span service-desc]
+                         :else
+                         [:span {:style {:color colors/gray650}}
+                          (tr [:service-search :no-description])])
         service-companies (cond
                             (and service-companies companies) (merge service-companies companies)
                             (and service-companies (empty? companies)) service-companies
                             :else companies)
-        spacer-margin {:margin "0.5rem 0 1.5rem 0"}]
-    [:div (stylefy/use-style style/result-card-new)
-     [:div (stylefy/use-sub-style style/result-card-new :header)
-      [:h3.result-title {:style {:margin "1rem 0"}}
-       name]
+        transport-types (map #(tr [:enums ::t-service/transport-type %]) transport-type)]
+    [:div (stylefy/use-style style/result-card)
+     [:div (stylefy/use-sub-style style/result-card :header)
```

Comment:
Even more invokations of stylefy/use-sub-style style/result-card, perhaps a helper function would help? Would it make reading harder as reader must then be able to know what the helper does, instead of just reading this invocation? I don't know.

Reply:
Yea it seems quite repetetive, but I don't see what we'd win by implementing a helper function that still needs to get the keyword as an argument.